src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -195,5 +233,111 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
+                        BrowsableAttribute browsableAttr = null;
+                        foreach (Attribute attr in field.GetCustomAttributes(typeof(BrowsableAttribute), false))
+                        {
+                            browsableAttr = attr as BrowsableAttribute;
+                        }
+
+                        if (browsableAttr == null || browsableAttr.Browsable)
+                        {
+                            object value = null;
+
+                            try
+                            {
+                                if (field.Name != null)
+                                {
+                                    value = Enum.Parse(_type, field.Name);
+                                }
+                            }
+                            catch (ArgumentException)
+                            {
+                                // Hmm, for some reason, the parse threw. Let us ignore this value.
+                            }
+
+                            if (value != null)
+                            {
+                                objValues.Add(value);
+                            }
+                        }
+                    }
+
+                    IComparer comparer = Comparer;
+                    if (comparer != null)
+                    {
+                        objValues.Sort(comparer);
+                    }
+                }
+
+                Array arr = (objValues != null) ? objValues.ToArray() : null;
+                _values = new StandardValuesCollection(arr);
+            }
+            return _values;
+        }
+
+        /// <internalonly/>
+        /// <devdoc>
+        ///    <para>Gets a value indicating whether the list of standard values returned from
+        ///    <see cref='System.ComponentModel.TypeConverter.GetStandardValues'/> 
+        ///    is an exclusive list using the specified context.</para>
+        /// </devdoc>
+        public override bool GetStandardValuesExclusive(ITypeDescriptorContext context)
+        {
+            return !_type.GetTypeInfo().IsDefined(typeof(FlagsAttribute), false);
+        }
+
+        /// <internalonly/>
+        /// <devdoc>
+        ///    <para>Gets a value indicating
+        ///       whether this object
+        ///       supports a standard set of values that can be picked
+        ///       from a list using the specified context.</para>
+        /// </devdoc>
+        public override bool GetStandardValuesSupported(ITypeDescriptorContext context)
+        {
+            return true;
+        }
+
+        /// <internalonly/>
+        /// <devdoc>
+        ///    <para>Gets a value indicating whether the given object value is valid for this type.</para>
+        /// </devdoc>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1803:AvoidCostlyCallsWherePossible")]  // Keep call to Enum.IsDefined
+        public override bool IsValid(ITypeDescriptorContext context, object value)
```

Comment:
Better link: https://msdn.microsoft.com/en-us/library/ms182260(v=vs.140).aspx. We can remove the `SuppressMessage` for this one - it's not an active rule anymore